src/ONGR/DemoMagentoBundle/Magento/Cart.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\DemoMagentoBundle\Magento;
+
+use ONGR\ElasticsearchBundle\ORM\Manager;
+use ONGR\MagentoConnectorBundle\Document\ProductDocument;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+/**
+ * Handles with cart.
+ */
+class Cart extends AbstractMagentoSync implements \Countable
+{
+    /**
+     * Name of the cookie where cart data is saved.
+     */
+    const CART_DATA_COOKIE_NAME = 'ongr_cart';
+
+    /**
+     * Parameter name for syncing with magento.
+     */
+    const CART_DATA_SYNC_PARAM_NAME = 'OngrProducts';
+
+    /**
+     * Parameter name for error list.
+     */
+    const CART_ERROR_LIST_PARAM_NAME = 'e';
+
+    /**
+     * Path to cart checkout in magento.
+     */
+    const CHECKOUT_PATH = '/checkout';
+
+    /**
+     * @var UrlGeneratorInterface
+     */
+    private $router;
+
+    /**
+     * @var Manager
+     */
+    private $manager;
+
+    /**
+     * @var array
+     */
+    private $cartContent;
+
+    /**
+     * @return string
+     */
+    public function getCheckoutUrl()
+    {
+        return $this->getMagentoUrl() . self::CHECKOUT_PATH;
+    }
+
+    /**
+     * Constructs request to update cart in magento.
+     *
+     * @return RedirectResponse
+     */
+    public function getUpdateResponse()
+    {
+        $url = $this->getMagentoUrl() . '?'
+            . http_build_query(
+                [
+                    self::CART_DATA_SYNC_PARAM_NAME => $this->getCartContent(),
+                    self::MAGENTO_BACK_URL_PARAM_NAME => $this->getBackUrl(),
+                ]
+            );
+
+        return new RedirectResponse($url);
+    }
+
+    /**
+     * Url magento should redirect after adding products.
+     *
+     * @return string
+     */
+    private function getBackUrl()
+    {
+        return $this->getRouter()->generate('ongr_cart', [], UrlGeneratorInterface::ABSOLUTE_URL);
+    }
+
+    /**
+     * Gets documents of cart products.
+     *
+     * @return array
+     */
+    public function getCartDocuments()
+    {
+        $cartContents = $this->getCartContent();
+        $repository = $this->getManager()->getRepository('ONGRMagentoConnectorBundle:ProductDocument');
+
+        $documents = [];
+
+        foreach ($cartContents as $id => $quantity) {
+            $documents[] = ['document' => $repository->find($id), 'quantity' => $quantity];
```

Comment:
I think this could be suboptimal, maybe this approach is better?

```
 public function getDocumentsByIds($ids)
    {
        $search = new Search();
        $search->addFilter(new IdsFilter($ids));
        /** @var DocumentIterator $results */
        $results = $this->providerManager->getRepository($this->repositoryType)->execute($search);

        return $results;
    }
```